common/src/main/java/org/apache/falcon/entity/ProcessHelper.java

```
@@ -185,4 +187,36 @@ private static void validateProcessInstance(Process process, Date instanceTime,
         }
         return result;
     }
+
+    public static Validity getClusterValidity(Process process, String clusterName) throws FalconException {
+        org.apache.falcon.entity.v0.process.Cluster cluster = getCluster(process, clusterName);
+        if (cluster == null) {
+            throw new FalconException("Invalid cluster: " + clusterName + " for process: " + process.getName());
+        }
+        return cluster.getValidity();
+    }
+
+    public static Sla getSLA(String clusterName, Process process) {
+        Cluster cluster = getCluster(process, clusterName);
+        return cluster != null ? getSLA(cluster, process) : null;
```

We should throw an exception here too, saying the cluster is invalid.